dev/com.ibm.ws.messaging.comms.client/src/com/ibm/ws/sib/netty/jfapchannel/NettyJMSClientHandler.java

```diff
@@ -0,0 +1,144 @@
+
+	}
+
+	@Override
+	protected void channelRead0(ChannelHandlerContext ctx, WsByteBuffer msg) throws Exception {
+		if (tc.isEntryEnabled())
+			SibTr.entry(this, tc, "channelRead0", ctx.channel());
+
+		Attribute<Connection> attr = ctx.channel().attr(CONNECTION_KEY);
+		Connection connection = attr.get();
+
+		if (connection != null) {
+			IOReadCompletedCallback callback = connection.getReadCompletedCallback();
+			IOReadRequestContext readCtx = connection.getReadRequestContext();
+			NetworkConnection networkConnection = connection.getNetworkConnection();
+			if(
+					callback instanceof NettyConnectionReadCompletedCallback && 
+					readCtx instanceof NettyIOReadRequestContext && 
+					networkConnection instanceof NettyNetworkConnection) {
+				((NettyConnectionReadCompletedCallback)callback).readCompleted(msg, readCtx, (NettyNetworkConnection)networkConnection);
+			}else {
+				if (TraceComponent.isAnyTracingEnabled() && tc.isWarningEnabled()) {
+        			SibTr.warning(tc, "channelRead0: Something's wrong. Callback, network connection, or read context is not netty specific. This shouldn't happen.", new Object[] {connection, callback, readCtx, networkConnection});
+                }
+        		exceptionCaught(ctx, new NettyException("Illegal callback type for channel."));
+        		return;
+			}
+
+		}else {
+			if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+				SibTr.warning(tc, "channelRead0: could not associate an incoming message with a Connection. Message will be ignored and channel will be closed.", new Object[] {ctx.channel()});
+				ctx.close();
+			}
+		}
+		if (tc.isEntryEnabled())
+			SibTr.exit(this, tc, "channelRead0", ctx.channel());
+	}
+
+	@Override
+	public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+		if (tc.isEntryEnabled())
+			SibTr.entry(this, tc, "channelInactive", ctx.channel());
+
+		// TODO: Check how to manage inactive channels appropriately
+		Connection connection = ctx.channel().attr(CONNECTION_KEY).get();
+		
+		if (tc.isEntryEnabled() && tc.isDebugEnabled()) {
+			SibTr.debug(this, tc, "Fired for connection which is closed?: " + connection.isClosed() + " closedDeffered?: " + connection.isCloseDeferred(), new Object[] {ctx.channel(), connection});
+		}
+		ctx.channel().attr(CONNECTION_KEY).set(null);
+		ctx.close();
+
+		if (tc.isEntryEnabled())
+			SibTr.exit(this, tc, "channelInactive", ctx.channel());
+	}
+
+	@Override
+	public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+		if (tc.isEntryEnabled())
+			SibTr.entry(this, tc, "exceptionCaught", new Object[] {ctx.channel(), cause});
+
+		Attribute<Connection> attr = ctx.channel().attr(CONNECTION_KEY);
+		Connection connection = attr.get();
+
+		if (connection == null) {
+			if (tc.isEntryEnabled() && tc.isDebugEnabled())
+				SibTr.debug(this, tc, "Found null connection for JMS Channel. Closing channel and moving cause through pipeline.", ctx.channel());
```

Comment:
I would have though this was unexpected? Might be more than a 'debug' level message?